src/sinks/splunk_hec/logs/config.rs

```diff
@@ -54,23 +64,35 @@ pub struct HecLogsSinkConfig {
     ///
     /// [splunk_field_index_docs]: https://docs.splunk.com/Documentation/Splunk/8.0.0/Data/IFXandHEC
     #[serde(default)]
+    #[configurable(metadata(docs::examples = "field1", docs::examples = "field2"))]
     pub indexed_fields: Vec<String>,
 
     /// The name of the index where to send the events to.
     ///
     /// If not specified, the default index is used.
+    #[configurable(metadata(docs::examples = "{{ host }}", docs::examples = "custom_index"))]
     pub index: Option<Template>,
```

Comment:
seems like this shouldn't be an Option and we should instead provide a default?
Or perhaps the wording "the default index is used" needs to be changed, if it isn't something vector is defaulting, but Splunk is.

Reply:
It is possible to set a default index in Splunk when creating an HTTP endpoint. I'd love to be able to link to some Splunk docs about this, but am unable to find anything useful.

![image](https://user-images.githubusercontent.com/192679/217785417-dc4f0688-655f-409c-8e5b-94500be55ffd.png)

Reply:
I've updated the wording a little to indicate that the default index is setup within Splunk.